### PR TITLE
feat: /review-learn — extract lessons from bot PR reviews (#930)

### LINKED ISSUE #930: feat: /review-learn skill — extract lessons from bot PR reviews (Pipeline 6, Phase 2)
## Summary

New skill that extracts lessons from bot review comments (CodeRabbit, GCA) post-merge. This is Phase 2 of the Bot Review Feedback Loop (Proposal 096 / pending ADR promotion).

## How it works

```bash
totem review-learn <pr-number>
```

1. Fetch all bot review comments on the PR
2. Filter to comments that were **fixed** (identified by reply text, emoji reaction, or commit addressing the file/line)
3. Generate lesson candidates from the structured findings
4. Deduplicate against existing lessons via vector search
5. Present for interactive selection (same UX as `totem extract`)
6. Write lessons with YAML frontmatter per ADR-070 (`lifecycle: nursery`)

Can also integrate into `/postmerge` workflow.

### Bot-specific parsing needed

- **CodeRabbit:** severity emojis, category tags, file+line, collapsible body
- **GCA:** priority SVGs, inline `suggestion` blocks, convention references

## Dependencies

- #929 — unified lesson schema (lessons need `lifecycle: nursery` frontmatter)
- #925 — `/review-reply` skill (Phase 1 — triage UX comes first)

## References

- Proposal 096: Bot Review Feedback Loop (Phases 1-4)
- ADR-058: Multi-Pipeline Confidence Engine (this is Pipeline 6)
- ADR-070: Unified Lesson Schema

## Milestone
1.6.0 — Pipeline Maturity

### PR DESCRIPTION
**Pull request by satur8d**
## Summary

- **New command**: `totem review-learn <pr-number>` — extracts lessons from resolved bot review comments on merged PRs
- **Bot parsers**: CodeRabbit (severity emojis, HTML `<details>` unwrapping) and GCA (SVG priorities, suggestion blocks)
- **Conservative resolution filter**: only learns from comments the developer explicitly fixed (pushback = skip)
- **Specialized LLM prompt**: enforces `lifecycle: nursery` per ADR-070
- **Full pipeline reuse**: dedup, interactive selection, lesson writing from `extract.ts`
- **Security**: `wrapUntrustedXml` for all bot content, custom secrets DLP
- **Strategy submodule update**: proposal lifecycle reorganization
- **72 new tests** (1,778 total across core + cli + mcp)

## The Feedback Loop

This closes the bot → lesson → rule → enforcement loop:
1. Bot reviews PR, posts findings
2. Developer fixes findings, replies "fixed"
3. `totem review-learn` extracts lessons from the fixed findings
4. `totem compile` turns lessons into deterministic rules
5. `totem lint` catches the pattern next time — no bot needed

## Test plan

- [x] Bot detection: CR, GCA, human (3 tests)
- [x] Severity parsing: CR emojis, GCA SVGs (4 tests)
- [x] HTML unwrapping: `<details>`, `<summary>`, comments (4 tests)
- [x] Resolution filter: positive signals, pushback rejection, edge cases (8 tests)
- [x] Prompt assembly: findings, XML wrapping, dedup context, truncation (7 tests)
- [x] Command guards: merged-only, no bots, no resolved findings (6 tests)
- [x] Template validation: nursery lifecycle, empty array fallback (6 tests)
- [x] Shield: PASS (0 findings)

Closes #930

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/cli/src/commands/review-learn.ts [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+    return;
+  }
+  log.info(TAG, `Found ${botComments.length} bot comment(s)`);
+
+  // 6. Group into threads
+  const threads = groupIntoThreads(botComments);
+  log.info(TAG, `Grouped into ${threads.length} bot review thread(s)`);
+
+  // 7. Apply resolution filter
+  const findings = extractResolvedBotFindings(threads);
```
[the remark this answers is not in the document]
Comment (satur8d) [reply packages/cli/src/commands/review-learn.ts]: Fixed — threads now grouped with ALL comments first, then filtered to bot-started threads. Human replies are preserved for resolution detection.
